### PR TITLE
ZMS-464 Install/configure/document master/slave SSDB cluster

### PR DESCRIPTION
**Pull request by gordyt**
This is a first-pass at the documentation.  Am submitting this to get feedback.  Note that this PR is based off of Travis' PR.  I suspect the best place to link this in will be from `ephemeraldata.adoc`.  Will do so and update the PR once Travis' request has  been merged.

### CODE REVIEW

File: ssdb-master-slave-replication.adoc [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+=== Keepalived configuration, Designated (Initial) Master
+
+
+`/etc/keepalived/keepalived.conf`
+
+The key configuration items to note are:
+
+- `state` -  State is set to `BACKUP` for _both_ the designated
+  (initial) master and backup servers.  In this scenario, the
+  `prioriy` is used to negotiate which server will assume `MASTER`
```
Comment (tmclane) [inline ssdb-master-slave-replication.adoc]: prioriy => priority?

File: ssdb-master-slave-replication.adoc [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+master server goes down, `Keepalived` will detect that failure and
+promote the slave, or backup, server to master.  `Keepalived` will
+maintain a https://en.wikipedia.org/wiki/Virtual_IP_address[Virtual
+IP address] on whichever server is the current master.
+
+The {product-name} will be configured such that the
+`zimbraEphemeralDataURL` will bind to the _Virtual IP address_ that is
+being maintained by `Keepalived`.  Once the installation and
+configuration has been completed, follow the instructions in
+link:ephemeraldata.adoc[Ephemeral Data] to update the `zimbraEphemeralDataURL` accordingly.
```
Comment (fatkudu) [inline ssdb-master-slave-replication.adoc]: s/zimbraEphemeralDataURL/zimbraEphemeralBackendURL/

File: ssdb-master-slave-replication.adoc [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+
+`Keepalived` will also be installed on these two servers.  Each
+`Keepalived` instance will monitor the other.  In the event that the
+master server goes down, `Keepalived` will detect that failure and
+promote the slave, or backup, server to master.  `Keepalived` will
+maintain a https://en.wikipedia.org/wiki/Virtual_IP_address[Virtual
+IP address] on whichever server is the current master.
+
+The {product-name} will be configured such that the
+`zimbraEphemeralDataURL` will bind to the _Virtual IP address_ that is
```
Comment (fatkudu) [inline ssdb-master-slave-replication.adoc]: s/zimbraEphemeralDataURL/zimbraEphemeralBackendURL/

File: ssdb-master-slave-replication.adoc [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+The key configuration items to note are:
+
+- `state` -  State is set to `BACKUP` for _both_ the designated
+  (initial) master and backup servers.  In this scenario, the
+  `prioriy` is used to negotiate which server will assume `MASTER`
+  status initially.
+- `nopreempt` - In the event that the master server fails and the
+  backup server is promoted to master, this configuration directive
+  will keep the original master from reclaiming that role.  This is
+  required because it likely be stale.  In this case, when it comes
```
Comment (fatkudu) [inline ssdb-master-slave-replication.adoc]: s/it likely be/it will likely be/ ?

**Comment by fatkudu** [comment]
`+1` apart from minor comments already made.

**Comment by gordyt** [comment]
**Note:** Am closing this PR.  Will create new document that discusses SSDB replication and scaling details, along with specific recommendations.

**Pull request closed without merging**